library/src/main/java/com/mindera/skeletoid/logs/appenders/LogFileAppender.java

```diff
@@ -72,8 +71,34 @@
      * @param fileName Log filename
      */
     public LogFileAppender(String tag, String fileName) {
-        TAG = tag;
+        if (tag == null || fileName == null) {
+            throw new IllegalArgumentException("TAG and fileName cannot be null");
+        }
+
         LOG_FILE_NAME = fileName + ".log";
+
+        if (isFilenameValid(LOG_FILE_NAME)) {
+            throw new IllegalArgumentException("Invalid fileName");
+        }
+
+        TAG = tag;
+    }
+
+    /**
+     * Check if fileName is valid
+     *
+     * @param fileName fileName
+     * @return true if it is, false if not
+     */
+    protected boolean isFilenameValid(String fileName) {
+        File f = new File(fileName);
+        try {
+            f.getCanonicalPath();
+            f.deleteOnExit();
```

Comment:
It's to validate if the file/path can be created. The delete part I think is not necessary.